src/js/ripple/remote.js

```
@@ -373,16 +377,49 @@ Remote.prototype._trace = function() {
   }
 };
 
+
+
+// Generate KeyPair from given seed. 
+Remote.prototype.generateKeyPair = function(secret, id) {
+  try {
+        var seed = Seed.from_json(secret);
+        var key  = seed.get_key(id);
+        return key;
+      } catch(e) {
+        return false;
+      }
+};
+
+// cache keypairs.
+Remote.prototype.setKeyPair = function(account, key) {
+  this.keyPairs[account] = key;
+}
+
 /**
  * Store a secret - allows the Remote to automatically fill
  * out auth information.
  *
  * @param {String} account
  * @param {String} secret
+ * @param {String|Number} account or index number.
  */
 
-Remote.prototype.setSecret = function(account, secret) {
+Remote.prototype.setSecret = function(account, secret, id) {
   this.secrets[account] = secret;
+
+  // try to cache keypairs, so we don't need to re-generate it for each tx.
+
+  // id could be RegularKey or index number of Account Family, 
+  // if ommited assume using Masterkey.
+  if (typeof id == 'undefined') id = account; 
+
+  var key = this.generateKeyPair(secret, id);
+
+  if (!key) {
+    this.emit('error', new RippleError('secretInvalid'));
```

I think I'd rather let this throw